service/src/main/kotlin/app/cash/backfila/service/runner/statemachine/BatchAwaiter.kt

```diff
@@ -57,6 +57,34 @@ class BatchAwaiter(
               backfillRunner.runBatchBackoff.addMillis(backfillRunner.metadata.extraSleepMs)
             }
           }
+
+          if (response.remaining_batch_range != null) {
+            // We have a remaining_batch_range, continue the batch.
+            val remaining = batch.newBuilder().batch_range(response.remaining_batch_range).build()
```

Comment:
do you have a test where the remaining batch fails and gets retried? I could see us regressing and retrying the original or something..

Reply:
Good point

Reply:
Ok added.

Reply:
And it found a bug by the way... It did retry the original. No longer.

Reply:
oh damn I didn't even see it, nice